Mergin/switch_sources_dialog.py

```diff
@@ -0,0 +1,202 @@
+        self.selectQgisProject.setStorageMode(QgsFileWidget.StorageMode.SaveFile)
+        self.selectQgisProject.fileChanged.connect(self.qgis_project)
+
+    def qgis_project(self, path: str) -> None:
+        self.lqgsproject_file.setText(path)
+
+
+class ProjectUsePostgreConfigWizard(QWizard):
+    """Wizard for changing project layer sources from GPKG to Postgre DB."""
+
+    def __init__(self, iface: QgisInterface, parent=None):
+        """Create a wizard"""
+        super().__init__(parent)
+
+        self.iface = iface
+        self.setWindowTitle("Create project with layers from Postgre Database")
+
+        self.connections: typing.List[Connection] = []
+
+        self.qgis_project = QgsProject.instance()
+
+        self.start_page = DBSyncConfigSelectionPage(self)
+        self.setPage(DBSYNC_PAGE, self.start_page)
+
+        self.qgsproject_page = QgsProjectSelectionPage(self)
+        self.setPage(QGS_PROJECT_PAGE, self.qgsproject_page)
+
+    def accept(self) -> None:
+        self.read_connections(self.start_page.field("db_sync_file"))
+
+        self.convert_gpkg_layers_to_postgis_sources(self.qgsproject_page.field("qgis_project"))
+
+        return super().accept()
+
+    def read_connections(self, path: str) -> None:
+        if path:
+            with open(path, mode="r", encoding="utf-8") as stream:
+                config = yaml.safe_load(stream)
+
+            self.connections = []
+            invalid_connections_info = []
+
+            for conn in config["connections"]:
+                connection = Connection(conn["driver"], conn["conn_info"], conn["modified"], conn["sync_file"])
+                if connection.valid:
+                    self.connections.append(connection)
+                else:
+                    invalid_connections_info.append(conn["conn_info"])
+
+            if invalid_connections_info:
+                self.iface.messageBar().pushMessage(
+                    "Mergin",
+                    f"Cannot connect to following databases: {'; '.join(invalid_connections_info)}.",
+                    level=Qgis.Critical,
+                    duration=0,
+                )
+        else:
+            self.connections = []
+
+    def new_project_parent_folder(self) -> str:
+        return os.path.dirname(self.qgsproject_page.field("qgis_project"))
+
+    def convert_gpkg_layers_to_postgis_sources(self, result_qgsproject_path: str):
+        update_project = QgsProject()
+        update_project.read(self.qgis_project.fileName())
+
+        project_layers = update_project.mapLayers()
+
+        layer: QgsMapLayer
+
+        for layer_id in project_layers:
+            layer = update_project.mapLayer(layer_id)
+
+            for dbsync_connection in self.connections:
+                if (
+                    layer.dataProvider().name() == "ogr"
+                    and dbsync_connection.sync_file in layer.dataProvider().dataSourceUri()
```

Comment:
shouldn't we do a better check for the URI match? e.g. if the sync_file is "data.gpkg" and there's also "backup/data.gpkg" file in the project, or maybe the QGIS project is also referencing a file outside of the MM project directory?